compiler/optimizer/LocalOpts.cpp

```diff
@@ -6932,183 +6928,108 @@ int32_t TR_InvariantArgumentPreexistence::perform()
 
       if (argInfo)
          {
-         if (trace())
-            traceMsg(comp(), "PREX:    Populating parmInfo of inlined method %s from argInfo %p\n", feMethod->signature(trMemory()), argInfo);
+         traceIfEnabled("PREX:    Populating parmInfo of inlined method %s from argInfo %p\n", feMethod->signature(trMemory()), argInfo);
 
          TR_PrexArgInfo *argInfo = comp()->getCurrentInlinedCallArgInfo();
-         int32_t index = 0;
+         index = 0;
          for (TR::ParameterSymbol *p = parms.getFirst(); p != NULL; p = parms.getNext(), index++)
             {
             TR_PrexArgument *arg = argInfo->get(index);
             ParmInfo &parmInfo = _parmInfo[index];
-            if (arg)
+            TR_OpaqueClassBlock *clazzFromMethod = parmInfo.getClass();
+
+            if (!parmInfo.isInvariant())
                {
-               if (trace())
-                  traceMsg(comp(), "PREX:      Parm %d is arg %p parmInfo %p\n", index, arg, &parmInfo);
+               // Clear non-invariant arg info
+               argInfo->set(index, NULL);
+               traceIfEnabled("PREX:       parm %d is not invariant\n", index);
+               continue;
                }
-            else
+
+            if (parmInfo.hasKnownObjectIndex() || !clazzFromMethod)
```

Comment:
this deserves a trace of some kind given the early return to loop head...

Reply:
Trace added.